### PR TITLE
Improve AI attachment validation and CloudWatch metrics integration

### PR DESCRIPTION
**Pull request by andtankian**
### 🧾 Summary

Streamlines AI attachment validation and CloudWatch metrics integration by removing unused interface fields and improving error logging and validation prompts.

### 🧩 Context

**Why this change?** Clean up CloudWatch metrics integration and improve AI validation messaging consistency

**Problem it solves:** Removes unused fields from CloudWatchMetricData interface and fixes duplicated validation responses in error logs

### 🧱 Scope of this PR

**This PR includes:**

- [ ] Bug fixes
- [x] Refactoring
- [ ] New features
- [ ] Documentation updates
- [x] Configuration changes

**Intentionally excluded:** Additional CloudWatch metrics enhancements (will be addressed in future iterations)

### 🧪 How to test

1. Run the test suites to verify all changes pass:
   ```bash
   pnpm nx test shared-cloudwatch-metrics
   pnpm nx test methodologies-bold-rule-processors-mass-id-document-manifest-data
   ```

2. Verify AI attachment validation still works correctly:
   - Run methodology processing with invalid attachments
   - Check that error messages are clear and not duplicated
   - Confirm CloudWatch metrics are recorded properly (if enabled)

3. Test environment configuration:
   ```bash
   pnpm nx test --testNamePattern="should handle test environment"
   ```

---

### ✅ Pre-merge Checklist

- [x] Code follows project style guidelines and naming conventions
- [x] All new code has appropriate test coverage
- [x] Linter warnings and errors have been resolved
- [x] No debugging code or console logs remain
- [x] TypeScript strict mode compliance verified
- [x] Tests pass locally and cover edge cases
- [x] Breaking changes documented with migration guide
- [x] Conventional commit format used for all commits
- [x] PR title and description follow project standards

## Summary by CodeRabbit

* **New Features**
  * Refined AI attachment validation messaging and clarified quantity validation guidance for certificates.
* **Refactor**
  * Reduced and hardened monitoring payloads so metric-recording failures no longer disrupt processing.
* **Chores**
  * Updated development/test environment variables and source URL.
* **Tests**
  * Adjusted tests to align with new validation messages, reduced metric payloads, and updated AI validation behavior.
* **Breaking/Integrations**
  * AI validation request/response shapes changed — integrations may require updates.

### CODE REVIEW

File: libs/methodologies/bold/rule-processors/mass-id/document-manifest-data/src/document-manifest-data.processor.ts [hunk 1]
```diff
@@ -248,17 +248,14 @@ export class DocumentManifestDataProcessor extends ParentDocumentRuleProcessor<R
 
       if (aiValidationResult.isValid === false) {
         logger.warn(
-          `AI validation failed for document manifest type ${this.documentManifestType} (${attachmentPath}): ${aiValidationResult.validationResponse}`,
+          `AI attachment validation failed for document manifest type ${this.documentManifestType} (${attachmentPath}): ${aiValidationResult.validationResponse} - ${aiValidationResult.validationResponse}`,
         );
```
[the remark this answers is not in the document]
Comment (andtankian) [reply libs/methodologies/bold/rule-processors/mass-id/document-manifest-data/src/document-manifest-data.processor.ts]: You're right! I fixed here e569eb9

File: libs/methodologies/bold/rule-processors/mass-id/document-manifest-data/src/document-manifest-data.processor.ts [hunk 2]
```diff
@@ -248,17 +248,14 @@
...
         const cloudWatchMetricsService = CloudWatchMetricsService.getInstance();
 
         if (cloudWatchMetricsService.isEnabled()) {
           await cloudWatchMetricsService.recordAIValidationFailure({
-            attachmentPath,
-            documentId: document.id,
             documentManifestType: this.documentManifestType,
-            validationResponse: aiValidationResult.validationResponse,
           });
         }
```
[the remark this answers is not in the document]
Comment (andtankian) [reply libs/methodologies/bold/rule-processors/mass-id/document-manifest-data/src/document-manifest-data.processor.ts]: c0ad6c1974559736f22a04a412cb72e7b30dfbff

File: .env-files/.env.test [hunk 3]
```diff
@@ -1,10 +1,16 @@
 ARTIFACT_CHECKSUM=0
-SOURCE_CODE_URL=https://carrot.eco
 SOURCE_CODE_VERSION=0
```
[the remark this answers is not in the document]
Comment (andtankian) [reply .env-files/.env.test:2]: b4400cd

[the remark this answers is not in the document]
Comment (andtankian) [reply .env-files/.env.test:2]: 39c6b83

**Review by sangalli — APPROVED** [review]
